app/helpers/application_helper/page_layouts.rb

```diff
@@ -165,6 +165,11 @@ def saved_report_paging?
     @sb[:pages] && @html && %i[reports_tree savedreports_tree cb_reports_tree].include?(x_active_tree)
   end
 
+  def show_search?
+    # Layouts with Advanced Search have the Search, too
+    controller.try(:show_searchbar?) || display_adv_search?
+  end
```

Comment:
I understand that you want to add this extra level of complexity, but then let's use it across the whole product. You're encapsulating the `display_adv_search?` here but it's still being used outside of this method in views.

I understand that you want to add complexity for the sake of readability, but then use it the right way and substitute it across the product, not just where you are introducing the search.

Reply:
Awesome, thank you very much for the comment :) So let's make some improvements.

Naming of the methods `show_search? and `show_searchbar?`- I am not sure if I understand properly why it is confusing. And what would you suggest?

Yes, I was focusing on readability. But what do you mean by "substitute it across the product" ? I am not sure. I think we really don't need `show_search?` anywhere else.

Reply:
See comment below.

Reply:
Also the `display_adv_search?` now is being called on at least 3 different abstraction level after your changes.

Reply:
Ok and how do you suggest to prevent this? I am sure there can be a better solution but I really need to call `display_adv_search?` somewhere, to keep the original logic of `_center_div_no_listnav` as much as possible and also because it makes sense to check this. 

I could call `if show_search? || display_adv_search?` directly in haml and then `display_adv_search?` would "stay" only in that haml as it was before and I could remove `display_adv_search?` from `show_search?`. Is this what you mean? But such a code in haml can be misleading, from my point of view.

Reply:
What about just calling `show_search?` everywhere?

Reply:
I feel I don't have good reasons for calling it everywhere. Maybe I just don't understand the problem properly. Could you help me more with this? Also I am not sure if I understand what do you mean by "everywhere".